### PR TITLE
patching clamp for one sided clamp

### LINKED ISSUE #75088: NVFuser produces wrong outputs for extreme values in clamp
(and possibly other operations)
```
def fn(x):
    x=x.clamp(min=1.)*.1
    return x

a=torch.tensor([1.,float('inf'), 2., float('inf')], device="cuda")
scripted = torch.jit.script(fn)
fn(a)
with torch.jit.fuser("fuser2"):
    for _ in range(10):
        scripted(a)
print(fn(a))
print(scripted(a))
```
produces
```
tensor([0.1000,    inf, 0.2000,    inf], device='cuda:0')
tensor([1.0000e-01, 3.4028e+37, 2.0000e-01, 3.4028e+37], device='cuda:0')
```

### PR DESCRIPTION
**Pull request by jjsjann123**
Fixes #75088 

The solution is just to avoid putting random value for non-specified clamp as pointed out in https://github.com/pytorch/pytorch/issues/75088#issuecomment-1093410036

### CODE REVIEW

**Comment by ngimel** [comment]
Cool, can you do a more systematic testing for extreme values? E.g. min/max look like they should be codegening fmin/fmax and thus handle nans correctly, but without systematic testing it's hard to be sure, there might be other ops like relu/hardtanh/hardsigmoid and the like that might or might not do correct things for nan/inf propagation, or produces 0-s in the non-differentiable points?

**Comment by jjsjann123** [comment]
> Cool, can you do a more systematic testing for extreme values? E.g. min/max look like they should be codegening fmin/fmax and thus handle nans correctly, but without systematic testing it's hard to be sure, there might be other ops like relu/hardtanh/hardsigmoid and the like that might or might not do correct things for nan/inf propagation, or produces 0-s in the non-differentiable points?

I think our unary/binary test do cover all these special numbers.
https://github.com/pytorch/pytorch/blob/1af7105a93d0cc2adfe34ed6c23df8d9c73c01c7/test/test_jit_cuda_fuser.py#L126-L130
https://github.com/pytorch/pytorch/blob/1af7105a93d0cc2adfe34ed6c23df8d9c73c01c7/test/test_jit_cuda_fuser.py#L544-L567
https://github.com/pytorch/pytorch/blob/1af7105a93d0cc2adfe34ed6c23df8d9c73c01c7/test/test_jit_cuda_fuser.py#L791-L796

`hardtanh` is not in our parser yet. But I do agree that we should probably having a generalized helper function for ternary ops as well. I'll update the tests then.

**Comment by ngimel** [comment]
Apparently, they weren't covering these special numbers, or these failure wouldn't have happened?
Another case where I see discrepancy between nvfuser and eager is when min is greater than max:
```
In [25]: def fn(x):
    ...:     x=x.clamp(min=1., max=0.5)*.1
    ...:     return x
    ...: 
    ...: a=torch.tensor([1.,float('inf'), 2., float('inf')], device="cuda")
    ...: scripted = torch.jit.script(fn)
    ...: fn(a)
    ...: with torch.jit.fuser("fuser2"):
    ...:     for _ in range(10):
    ...:         scripted(a)
    ...: print(fn(a))
    ...: print(scripted(a))
tensor([0.0500, 0.0500, 0.0500, 0.0500], device='cuda:0')
tensor([0.0500, 0.0500, 0.0500, 0.0500], device='cuda:0')

In [26]: print(fn(x.cuda()))
tensor([0.0500, 0.0500, 0.0500, 0.0500], device='cuda:0')

In [27]: print(scripted(x.cuda()))
tensor([0.1000, 0.1000, 0.1000, 0.0500], device='cuda:0')

In [28]: x
Out[28]: tensor([-0.3980, -0.2727,  0.1300,  2.0310])
```
Note, although this is arguably an error case, numpy, jax and torch eager all have the same behavior in this case, with nvfuser being an outlier.

**Comment by jjsjann123** [comment]
clamp is a `ternary` op which is not covered by unary/binary tests and that's what I was promising to update in our tests.

`x=x.clamp(min=1., max=0.5)` That's a confusing case... I think we just need to add two lines to regulate the min/max range. I'll update that to aten logic... But if these cases are defined behavior, maybe we should update opinfo tests to have it backed up by CI?

**Comment by ngimel** [comment]
As far as NVFuser is concerned, clamp is a unary op with kwargs, NVFuser accepts only (Tensor, Scalar, Scalar) overload.

**Comment by ngimel** [comment]
And yes, adding a test for this behavior (since it's documented) is necessary, I wasn't able to find existing one. cc @mruberry

File: torch/csrc/jit/codegen/cuda/parser.cpp [hunk 1]
```diff
@@ -1138,14 +1138,34 @@ class IrParser {
...
-            Val* low = value_map.count(node->inputs()[1]->unique()) != 0
+            bool has_low = value_map.count(node->inputs()[1]->unique()) != 0;
+            Val* low = has_low
                 ? *value_map[node->inputs()[1]->unique()]
                 : IrBuilder::create<Double>(std::numeric_limits<float>::min());
-            Val* high = value_map.count(node->inputs()[2]->unique()) != 0
+            bool has_high = value_map.count(node->inputs()[2]->unique()) != 0;
+            Val* high = has_high
                 ? *value_map[node->inputs()[2]->unique()]
                 : IrBuilder::create<Double>(std::numeric_limits<float>::max());
```
Comment (ngimel) [inline torch/csrc/jit/codegen/cuda/parser.cpp:1148]: You still have these very misleading `high` and `low` assignments, you should be able to remove them?

**Review by ngimel — APPROVED** [review]

**Pull request closed without merging**